### ISSUE TITLE
additional constraints with look_ahead_regex, look_behind_regex don't work anymore

### ISSUE THREAD

**Issue by anantb** [Microsoft/prose#7]
Was playing with the new update in a hackathon. The below code results into learning failed if I pass a proper regex for `lookBehindRegex`, `matchingRegex`, and `lookAheadRegex`.

```
Microsoft.ProgramSynthesis.Extraction.Text.RegionLearner.Instance.LearnTopK(
    exampleConstraints, null, 1, lookBehindRegex, matchingRegex, lookAheadRegex);
```

I had played with the APIs a few months earlier, and in the previous version (`0.1.1-preview-1`), the below code worked fine and the resulting program would be consistent with `lookBehindRegex`, `matchingRegex`, and `lookAheadRegex`:

```
Microsoft.ProgramSynthesis.Extraction.Text.Learner.Instance.LearnTopKRegion(                
    positiveExamples, negativeExamples, null, 1, lookBehindRegex, matchingRegex, lookAheadRegex);
```

**Comment by apskim** [comment]
Hi @anantb! Can you please give a complete code snippet with the constraints?

**Comment by anantb** [comment]
Yes, will send you a reproducible code snippet by tomorrow.

**Comment by anantb** [comment]
Hi @apskim.

Here is the Snippet that breaks:

```
using System;
using System.Collections.Generic;
using System.Linq;
using Microsoft.ProgramSynthesis.Extraction.Text;
using Microsoft.ProgramSynthesis.Extraction.Text.Semantics;
using Microsoft.ProgramSynthesis.Wrangling.Constraints;
using System.Text.RegularExpressions;

namespace TextExtraction
{
	class MainClass
	{
		public static void Main(string[] args)
		{
			var input = RegionLearner.CreateStringRegion(
				"64.242.88.10 - - [07/Mar/2004:16:05:49 -0800] \"GET /twiki/bin/edit/Main/Double_bounce_sender?topicparent=Main.ConfigurationVariables HTTP/1.1\" 401 12846"
			);


			var examples = new[] {
				new CorrespondingMemberEquals<StringRegion, StringRegion>(input, input.Slice(0,12))
            };

			RegionProgram prog = null;
			IEnumerable<RegionProgram> progList = new List<RegionProgram>();

			Regex lookBehindRegex = null;
			lookBehindRegex = new Regex("\\s+\\-");

			progList = RegionLearner.Instance.LearnTopK(examples, null, 1, lookBehindRegex, null, null);
			if (progList.Count() > 0) {
				prog = progList.First();
			} else {
				Console.WriteLine("Error: Learning Failed");
			}

			var testInput = RegionLearner.CreateStringRegion(
					"64.242.88.10 - - [07/Mar/2004:16:06:51 -0800] \"GET /twiki/bin/rdiff/TWiki/NewUserTemplate?rev1=1.3&rev2=1.2 HTTP/1.1\" 200 4523"
			);
			StringRegion output = prog.Run(testInput);
			if (output == null)
			{
				Console.Error.WriteLine("Error: Extracting failed!");
				return;
			}
			Console.WriteLine("\"{0}\" => \"{1}\"", testInput, output);
		}
	}
}
```

This the the exception that it throws. It used to work  fine earlier.

```
System.ArgumentNullException
Value cannot be null.
Parameter name: evaluateLogLikelihood
```

**Comment by apskim** [comment]
Thanks, @anantb! This bug is already fixed in our internal repo, and we will push the fix to NuGet soon.
I will update and close the issue when it happens.

**Comment by anantb** [comment]
Thanks a lot! Really appreciate it.

**Comment by apskim** [comment]
**Upd:** we're moving to a fixed monthly release schedule (barring critical issues), so expect to see this fix in our next monthly update 😉

**Comment by anantb** [comment]
Perfect.

**Comment by apskim** [comment]
We just published PROSE 1.1.0 ([release notes](https://microsoft.github.io/prose/release-notes/)), which, among other things, fixes this bug.

**Comment by anantb** [comment]
Perfect -- works now.
